::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
: Copyright (c) 2017 Mirantis Inc., Enea AB and others.
:
: All rights reserved. This program and the accompanying materials
: are made available under the terms of the Apache License, Version 2.0
: which accompanies this distribution, and is available at
: http://www.apache.org/licenses/LICENSE-2.0
::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
From: Alexandru Avadanii <Alexandru.Avadanii@enea.com>
Date: Tue, 30 Jan 2018 01:23:54 +0100
Subject: [PATCH] salt-master-setup: Group APT install formulas

Instead of calling `apt install` for each salt formula package that
we miss, construct a list and install them all at once.

Signed-off-by: Alexandru Avadanii <Alexandru.Avadanii@enea.com>
---
 salt-master-setup.sh | 12 +++++++-----
 1 file changed, 7 insertions(+), 5 deletions(-)

diff --git a/salt-master-setup.sh b/salt-master-setup.sh
index 0dd3036..a0c6311 100755
--- a/salt-master-setup.sh
+++ b/salt-master-setup.sh
@@ -347,15 +347,17 @@ install_salt_formula_pkg()
           # Set essentials if FORMULAS_SALT_MASTER is not defined at all
           [ -z ${FORMULAS_SALT_MASTER+x} ] && declare -a FORMULAS_SALT_MASTER=("linux" "reclass" "salt" "memcached")
           for formula_service in "${FORMULAS_SALT_MASTER[@]}"; do
-              echo -e "\nConfiguring salt formula ${formula_service} ...\n"
+              echo -e "Configuring salt formula ${formula_service} ..."
               [ ! -d "${FORMULAS_PATH}/env/${formula_service}" ] && \
-                  if ! $SUDO apt-get install -y salt-formula-${formula_service}; then
-                    echo -e "\nInstall salt-formula-${formula_service} failed.\n"
-                    exit 1
-                  fi
+                  _FORMULAS_SALT_MASTER="${_FORMULAS_SALT_MASTER} salt-formula-${formula_service}"
+              # Create links first, install pkgs later
               [ ! -L "/srv/salt/reclass/classes/service/${formula_service}" ] && \
                   ln -sf ${FORMULAS_PATH}/reclass/service/${formula_service} /srv/salt/reclass/classes/service/${formula_service}
           done
+          if ! $SUDO apt-get install -qqq -y ${_FORMULAS_SALT_MASTER}; then
+            echo -e "\nInstall ${_FORMULAS_SALT_MASTER} failed.\n"
+            exit 1
+          fi
         ;;
       rhel)
         # TODO
